### ISSUE TITLE
Crash when using glyph ranges when using AddFonts

### ISSUE THREAD

**Issue by neclepsio** [AllenDang/cimgui-go#92]
There's a crash when using glyph ranges because in `extra_types.go` you define `Wchar` as `C.uint`, while `cimgui.h` defines it as `C.ImWchar32`.

Also, if `IMGUI_USE_WCHAR32` is not defined the type should be `C.ushort`. Looking at GitHub Actions and `cimgui.h`, it seems to me that `IMGUI_USE_WCHAR32` is indeed used.

Changing `Wchar` to `C.ImWchar` in `extra_types.go` the crash is not solved.

Changing `Wchar` to `C.ImWchar16` in `extra_types.go` __and__ `typedef ImWchar32 ImWchar` to `typedef ImWchar16 ImWchar` in `cimgui.h` the crash is solved. So it seems that there is a mismatch between `cimgui.h` and `imgui.a` (only testend on Windows).

**Comment by gucio321** [comment]
yah, but sadly we can't/shouldn't edit `cimgui.h` as it is drop-in code from upstream cimgui repo...

**Comment by neclepsio** [comment]
> yah, but sadly we can't/shouldn't edit `cimgui.h` as it is drop-in code from upstream cimgui repo...

Of course, that's why `imgui.a` should be compiled using `-DIMGUI_USE_WCHAR32`

**Comment by sonoro1234** [comment]
> 

Imgui defines can be changed in cimgui generation

**Comment by sonoro1234** [comment]
> > yah, but sadly we can't/shouldn't edit `cimgui.h` as it is drop-in code from upstream cimgui repo...
> 
> Of course, that's why `imgui.a` should be compiled using `-DIMGUI_USE_WCHAR32` (if my analysis is correct).

Yes because the cimgui fork in this repo is generated for wchar32

**Comment by gucio321** [comment]
ok, @neclepsio @sonoro1234 so as far as I understand, first of all we need to make our workflows to compile binaries with `-DIMGUI_USE_WCHAR32` cmake flag enabled.
After that we need to edit extra_types.go to make Wchar a valid type.
My question is, what's the valid type (instead of `C.uint`) for ImWchar?

**Comment by sonoro1234** [comment]
see https://github.com/cimgui/cimgui/blob/docking_inter/cimgui.h#L193
```c
typedef unsigned short ImWchar16;
typedef unsigned int ImWchar32;
typedef ImWchar16 ImWchar;
```
The last typedef comes from https://github.com/ocornut/imgui/blob/3482d4eccfc6fe68f1db134829caf908ac9e294e/imgui.h#L238 in case IMGUI_USE_WCHAR32 is defined.
```cpp
#ifdef IMGUI_USE_WCHAR32            // ImWchar [configurable type: override in imconfig.h with '#define IMGUI_USE_WCHAR32' to support Unicode planes 1-16]
typedef ImWchar32 ImWchar;
#else
typedef ImWchar16 ImWchar;
#endif

```
IMGUI_USE_WCHAR32 must be the same in cimgui generation and imgui compilation.

I am not sure about the go terminology for C types but:
with IMGUI_USE_WCHAR32 not defined, ImWchar could be `C.ushort`?
with IMGUI_USE_WCHAR32 defined as in https://github.com/AllenDang/cimgui-go/blob/main/cimgui/cimgui.h#L195 , ImWchar could be `C.uint`?

**Comment by neclepsio** [comment]
It should be `C.ImWchar`, I think the problem is the .h defines it as 32 bits while the .a is compiled as 16 bits.

**Comment by sonoro1234** [comment]
> It should be `C.ImWchar`, I think the problem is the .h defines it as 32 bits while the .a is compiled as 16 bits.

Yes, IMGUI_USE_WCHAR32 must be the same in cimgui generation and imgui compilation.

**Comment by gucio321** [comment]
ok, first of all, @neclepsio could you please post some reproducable example, since I cannot test the fix

**Comment by sonoro1234** [comment]
Perhaps using a very populated font which has codepoints  larger than 16bits?
One example can be found in https://github.com/ocornut/imgui/issues/4959

**Comment by gucio321** [comment]
@neclepsio this is probably the same problem as in #175 so dhould be fixed by #177
